src/app/Shared/Services/Report.service.tsx

```diff
@@ -80,21 +80,30 @@ export class ReportService {
         const isArchived = !isActiveRecording(recording);
         const isActivedStopped = isActiveRecording(recording) && recording.state === RecordingState.STOPPED;
         if (isArchived || isActivedStopped) {
-          this.reports.set(recording.reportUrl, report);
+          try {
+            sessionStorage.setItem(this.key(recording), report);
+          } catch (error) {
+            this.notifications.warning('Report Caching Failed', error.message);
+            sessionStorage.clear();
```

Comment:
If the session storage is full and throws an error, is the purpose of `sessionStorage.clear()` to make space for the report URL the next time the report is requested? 

https://developer.mozilla.org/en-US/docs/Web/API/Storage/setItem#exceptions

Reply:
Yes, I figure if the store is full then it's time to clear it out to make room for more cached items. It isn't very elegant, but I don't really want to implement a whole client-side LRU cache on top of sessionStorage. Since we don't know which other items in cache are oldest or least recently accessed, just that the cache is full and we couldn't fit the most recently requested thing, I just clear it out so that the next access can be cached again. It would be even better if I could clear the cache and re-`setItem` the new report into the cache, but this:

```
Particularly, in Mobile Safari (since iOS 5) it always throws when the user enters private mode. (Safari sets the quota to 0 bytes in private mode, unlike other browsers, which allow storage in private mode using separate data containers.)
```

makes that sound infeasible for at least some possible scenarios.

The Cryostat backend does some caching as well anyway, so this client-side cache is primarily just to avoid the network round trip latency of closing and reopening the UI element.